src/Lean/PrettyPrinter.lean

```diff
@@ -101,4 +98,37 @@ unsafe def registerParserCompilers : IO Unit := do
   ParserCompiler.registerParserCompiler ⟨`formatter, formatterAttribute, combinatorFormatterAttribute⟩
 
 end PrettyPrinter
+
+namespace MessageData
+
+open Lean PrettyPrinter Delaborator
+
+/--
+Turns a `MetaM FormatWithInfos` into a `MessageData` using `.ofPPFormat` and running the monadic value in the given context.
+Uses the `pp.tagAppFns` option to annotate constants with terminfo, which is necessary for seeing the type on mouse hover.
+-/
+def ofFormatWithInfos
+    (fmt : MetaM FormatWithInfos)
+    (noContext : Thunk Format := "<no context, could not generate MessageData>") : MessageData :=
```

Comment:
I think one-shot lazy values usually use `Unit -> _` instead of `Thunk _`, and I *believe* `Unit -> _` is a bit more efficient, but I've never measured it.
```suggestion
    (noContext : Unit -> Format := fun _ => "<no context, could not generate MessageData>") : MessageData :=
```

Reply:
Fixed!